pkg/agent/multicluster/stretched_networkpolicy_controller.go

```diff
@@ -0,0 +1,388 @@
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(cur interface{}) {
+				controller.processLabelIdentityAddOrUpdate(cur)
+			},
+			UpdateFunc: func(old, cur interface{}) {
+				controller.processLabelIdentityAddOrUpdate(cur)
+			},
+			DeleteFunc: func(old interface{}) {
+				controller.processLabelIdentityDelete(old)
+			},
+		},
+		resyncPeriod,
+	)
+	return controller
+}
+
+func (s *StretchedNetworkPolicyController) Run(stopCh <-chan struct{}) {
+	defer s.queue.ShutDown()
+
+	klog.InfoS("Starting Multi-cluster Controller", "controllerName", stretchedNetworkPolicyControllerName)
+	defer klog.InfoS("Shutting down Multi-cluster Controller", "controllerName", stretchedNetworkPolicyControllerName)
+	cacheSyncs := []cache.InformerSynced{s.podListerSynced, s.namespaceListerSynced, s.LabelIdentityListerSynced}
+	if !cache.WaitForNamedCacheSync(stretchedNetworkPolicyControllerName, stopCh, cacheSyncs...) {
+		return
+	}
+
+	for i := 0; i < defaultWorkers; i++ {
+		go wait.Until(s.worker, time.Second, stopCh)
+	}
+	<-stopCh
+}
+
+func (s *StretchedNetworkPolicyController) initLabelIDCache() {
+	labelIdentities, err := s.labelIdentityLister.List(labels.Everything())
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	if err == nil {
+		for _, labelIdentity := range labelIdentities {
+			s.labelIdentityCache[labelIdentity.Spec.Label] = labelIdentity.Spec.ID
+		}
+	}
+}
+
+// worker is a long-running function that will continually call the processNextWorkItem
+// function in order to read and process a message on the workqueue.
+func (s *StretchedNetworkPolicyController) worker() {
+	for s.processNextWorkItem() {
+	}
+}
+
+func (s *StretchedNetworkPolicyController) processNextWorkItem() bool {
+	obj, quit := s.queue.Get()
+	if quit {
+		return false
+	}
+	defer s.queue.Done(obj)
+
+	if p, ok := obj.(*v1.Pod); !ok {
+		s.queue.Forget(obj)
+		klog.Errorf("Expected type 'pod' in work queue but got object", "object", obj)
+	} else if err := s.syncPodClassifierFlow(p); err == nil {
+		s.queue.Forget(p)
+	} else {
+		// Put the item back on the workqueue to handle any transient errors.
+		s.queue.AddRateLimited(p)
+		klog.ErrorS(err, "Error syncing podClassifierFlowUpdateItem, requeuing", "podName", p.Name, "podNS", p.Namespace)
+	}
+	return true
+}
+
+// syncPodClassifierFlow will update labelToPods and podToLabel and update
+// PodClassifierFlow with its LabelIdentity, if available, or UnknownLabelIdentity.
+func (s *StretchedNetworkPolicyController) syncPodClassifierFlow(pod *v1.Pod) error {
+	podRef := getPodReference(pod)
+	podNS, err := s.namespaceLister.Get(pod.Namespace)
+	if err != nil {
+		return err
+	}
+	normalizedLabel := getNormalizedLabel(podNS.Labels, pod.Labels, podNS.Name)
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	oldNormalizedLabel, ok := s.podToLabel[podRef]
+	if ok && oldNormalizedLabel != normalizedLabel {
+		s.deleteLabelToPod(oldNormalizedLabel, pod)
+	}
+	if !ok || oldNormalizedLabel != normalizedLabel {
```

Comment:
There are three conditions that we should take care of:
1. `oldNormalizedLabel` exist and it's different than the new `normalizedLabel `.
        We need to delete and add/update cache map then update flows.
2. `oldNormalizedLabel` exist and it's the same as the new `normalizedLabel `.
        We only need to update flows.
3. `oldNormalizedLabel`  doesn't exist.
        We need add/update cache map then update flows.

I guess you think case 2 won't happen since we check whether labels have changed while handling Pod events. But maybe you miss the situation that `LabelIdentity` update events will also enqueue Pods even PodLabels are the same.